core/src/main/scala/io/projectglow/common/GenomicIntervalUtils.scala

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2019 The Glow Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.projectglow.common
+
+import scala.math.{max, min}
+
+object GenomicIntervalUtils {
+
+  /**
+   * A class representing a closed interval of longs that also accommodates an empty interval.
+   */
+  class LongInterval() extends Serializable {
```

Comment:
It seems like this should just be a
```scala
case class LongInterval(coordinates: Option[(Long, Long)])
```

Then you get `isSame` (which should just be `equals`), `isEmpty`, and `nonEmpty` for free.